src/BenchmarkDotNet/Configs/ManualConfig.cs

```diff
@@ -196,6 +196,10 @@ public ManualConfig AddJob(params Job[] newJobs)
         public ManualConfig AddHardwareCounters(params HardwareCounter[] newHardwareCounters)
         {
             hardwareCounters.AddRange(newHardwareCounters);
+            columnProviders.AddRange(
+                newHardwareCounters
+                    .Select(h => new MetricColumn(new HardwareCounterDescriptor(h)))
+                    .Select(c => c.ToProvider()));
```

Comment:
This can be done with a single Select.

Reply:
Also, this looks kind of strange to me. The column providers by default are added from `DefaultColumnProviders`. I think the hardware counters just need to be added to the metrics when the report is generated so it will be picked up automatically by the default `MetricsColumnProvider`.

Reply:
Hi @timcassell,

I checked and the DefaultColumnProviders add the Metrics columns, but they are not being used on the ManualConfig.cs unless the user calls ManualConfig.CreateMinimal(). So the following code in ManuaklConfig.cs may work for when they create an empty ManualConfig and add Hardware Counters? 


```cs       
        public ManualConfig AddHardwareCounters(params HardwareCounter[] newHardwareCounters)
        {
            hardwareCounters.AddRange(newHardwareCounters);
            AddColumnProvider(DefaultColumnProviders.Metrics);
            return this;
        }

Reply:
So the exporter already works when using default config? Or am I misunderstanding?

Reply:
Yes, it works. Only the XmlExporter was missing the Metrics data, that one needs the change to the file SummaryDto.cs.

The Exporter in a ManualConfig also works, but by calling AddHardwareCounters in a ManualConfig does not add the Counters Columns by default (to be printed and exported), so the columns will need to be added manually. The code above tries to solve this by adding the metric columns by default when Hardware Counters are added to a ManualConfig.